internal/xds/rbac/rbac_engine.go

```diff
@@ -131,18 +138,45 @@ func newEngine(config *v3rbacpb.RBAC) (*engine, error) {
 		}
 		policies[name] = matcher
 	}
+
+	auditLoggers, auditCondition, err := parseAuditOptions(config.GetAuditLoggingOptions())
+	if err != nil {
+		return nil, err
+	}
 	return &engine{
-		policies: policies,
-		action:   a,
+		policies:       policies,
+		action:         a,
+		auditLoggers:   auditLoggers,
+		auditCondition: auditCondition,
+		policyName:     policyName,
 	}, nil
 }
 
+func parseAuditOptions(opts *v3rbacpb.RBAC_AuditLoggingOptions) ([]audit.Logger, v3rbacpb.RBAC_AuditLoggingOptions_AuditCondition, error) {
+	if opts == nil {
+		return nil, v3rbacpb.RBAC_AuditLoggingOptions_NONE, nil
+	}
+	var auditLoggers []audit.Logger
+	for _, logger := range opts.LoggerConfigs {
+		auditLogger, err := buildLogger(logger)
+		if err != nil {
+			return nil, v3rbacpb.RBAC_AuditLoggingOptions_NONE, err
+		}
+		if auditLogger == nil {
+			continue
+		}
```

Comment:
Please add a comment here...  `// Audit logger is not registered, but is optional; skip`.

Or....should the `optional` check be done here instead of in `buildLogger`?

Is it proper that "optional" results in a failure in some situations, but not if the logger is just missing?

Reply:
Yes `is_optional` will only let us ignore unregistered loggers. We will document this behavior.